### PR TITLE
Actions

### PR DESCRIPTION
**Pull request by Boy0000**
Moved @AidanF86 actions from https://github.com/MineInAbyss/MineInAbyss/pull/76 as they should probably be in here?

### CODE REVIEW

File: geary-commons-papermc/src/main/kotlin/com/mineinabyss/geary/minecraft/actions/FreezeEntityAction.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package com.mineinabyss.geary.minecraft.actions
+
+import com.mineinabyss.geary.ecs.api.actions.GearyAction
+import com.mineinabyss.geary.ecs.api.entities.GearyEntity
+import org.bukkit.entity.Entity
+
+class FreezeEntityAction() : GearyAction() {
```
Comment (0ffz) [inline geary-commons-papermc/src/main/kotlin/com/mineinabyss/geary/minecraft/actions/FreezeEntityAction.kt:7]: Make freezeTicks configurable with a constructor parameter

File: geary-commons-papermc/src/main/kotlin/com/mineinabyss/geary/minecraft/actions/IgniteBlocksInRadiusAction.kt [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+            for (y in -intRadius..intRadius) {
+                for (z in -intRadius..intRadius) {
+                    val location = Location(
+                        entity.world,
+                        entity.location.x + x,
+                        entity.location.y + y,
+                        entity.location.z + z
+                    )
+                    if (location.block.isEmpty) {
+                        val belowLocation = Location(
```
Comment (0ffz) [inline geary-commons-papermc/src/main/kotlin/com/mineinabyss/geary/minecraft/actions/IgniteBlocksInRadiusAction.kt:26]: use `location.add(0.0, 1.0, 0.0)`

Also why's it called belowLocation if we are adding one? would it not be above?

File: geary-commons-papermc/src/main/kotlin/com/mineinabyss/geary/minecraft/actions/IgniteEntityAction.kt [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package com.mineinabyss.geary.minecraft.actions
+
+import com.mineinabyss.geary.ecs.api.actions.GearyAction
+import com.mineinabyss.geary.ecs.api.entities.GearyEntity
+import org.bukkit.entity.Entity
+
+class IgniteEntityAction(
+    private val seconds: Int = 7
```
Comment (0ffz) [inline geary-commons-papermc/src/main/kotlin/com/mineinabyss/geary/minecraft/actions/IgniteEntityAction.kt:8]: Use our TimeSpan class, call it length and do `length.inTicks`

File: geary-commons-papermc/src/main/kotlin/com/mineinabyss/geary/minecraft/actions/SelfDestructAction.kt [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+import com.mineinabyss.geary.ecs.api.actions.GearyAction
+import com.mineinabyss.geary.ecs.api.entities.GearyEntity
+import org.bukkit.entity.Damageable
+import org.bukkit.entity.Entity
+
+/**
+ * Self destructs (makes an explosion and kills) an entity
+ */
+class SelfDestructAction(
```
Comment (0ffz) [inline geary-commons-papermc/src/main/kotlin/com/mineinabyss/geary/minecraft/actions/SelfDestructAction.kt:11]: Do we need an action that's basically just two separate actions?

File: geary-commons-papermc/src/main/kotlin/com/mineinabyss/geary/minecraft/actions/IgniteBlocksInRadiusAction.kt (hunk 2 above)
Comment (AidanF86) [reply geary-commons-papermc/src/main/kotlin/com/mineinabyss/geary/minecraft/actions/IgniteBlocksInRadiusAction.kt]: Yeah mistype

**Comment by Boy0000** [comment]
max making a new PR

**Pull request closed without merging**